DTM serial port connection documentation (#4040)
nrfconnect/sdk-nrf: doc/nrf/dtm_testing.rst

.. _dtm_testing:

Direct Test Mode terminal connection
====================================

.. contents::
   :local:
   :depth: 2

> Follow the instructions in this section to connect with the DK which running the DTM to send commands and receive responses.

Reviewer: ```suggestion
Follow these instructions to connect with the DK running the DTM and send commands and receive responses.
```

> Follow the instructions in this section to connect with the DK which running the DTM to send commands and receive responses.

Reviewer 2: ```suggestion
Follow the instructions in this section to connect with the DK which is running the DTM to send commands and receive responses.
```

> The Bluetooth Low Energy DTM UART interface standard specifies:

Reviewer: ```suggestion
The Bluetooth Low Energy DTM UART interface standard specifies the following configuration:
```

* 8 data bits
* No parity
* 1 stop bit
* No hardware flow control
* A selection of bit rates from 9600 to 1000000, one of which must be supported by the DUT.
  It might be possible to run other bit rates by experimenting with parameters.

> 

Reviewer: Numbers below 10 spelled out: Eight data bits, One stop bit

.. note::
   The default bit rate of the DTM UART driver is 19200 bps, which is supported by most certified testers.

All command must be sent as two bytes Hex number, also responses have a format of two bytes Hex number.

> How to connect with RealTerm (Windows)

Reviewer: ```suggestion
Connect with RealTerm (Windows)
```

**************************************

The RealTerm start window:

.. figure:: /images/realterm.svg

>    :alt: RealTerm start window

Reviewer: Do we want to have a caption for the image?

> To test the DTM with the RealTerm follow these steps:

Reviewer: ```suggestion
To test the DTM with the RealTerm, complete the following steps:
```

> 1. On the **Display** tab set Display As to **Hex[space]**

Reviewer: ```suggestion
1. On the :guilabel:`Display` tab set :guilabel:`Display As` to :guilabel:`Hex[space]`.
```

.. figure:: /images/realterm_hex_display.png

>    :alt: Set the RealTerm display format

Reviewer: Do we want to have a caption for the image?

> 2. Go to the **Port** tab and configure a serial port parameters:

Reviewer: Open the :guilabel:`Port` tab and configure the serial port parameters:

   For example:

.. code-block::

   sudo minicom -D /dev/serial/by-id/usb-SEGGER_J-Link_000683580193-if00 -s

   The **-s** option is used to enter a minicom setup mode.

2. Configure the **minicom** terminal

   Configuration window:

.. figure:: /images/minicom_setup_window.png
   :alt: minicom configuration window

   Enter the Serial port setup and set the UART baudrate to 19200:

.. figure:: /images/minicom_serial_port.png
   :alt: minicom serial port settings

   Next enter the **Screen and keyboard** and press **S** on a keyboard to enable the **Hex Display**. Next press **Q** on a keyboard to enable the **Local echo**.

.. figure:: /images/minicom_terminal_cfg.png
   :alt: minicom terminal screen and keyboard settings

   The minicom is now configured for receiving data from the minicom.
   However it cannot be used for sending DTM commands.

3. Sending DTM commands

   This easiest way to send the DTM command is to use the **echo** with **-ne** options in the second terminal.
   You must encode data as Hex numbers (\xHH, byte with hexadecimal value HH (1 to 2 digits))

.. code-block::

   sudo echo -ne "<encoded command>" > <DTM serial port>

> For example (sending **Reset** command):

Reviewer: To send a **Reset** command, for example, run the following command:
